### DISCUSSION TITLE
how to update  the context in  activities

### DISCUSSION THREAD

**Discussion by 000luna000** [statelyai/xstate#3183] [General] [answered]
I want to listen to some events in activities, once these external events are triggered, I need to change the context.

**Comment by Andarist** [comment, marked as answer]
It would be best if you could share a pseudo-code of what you are trying to achieve.

If I understand the question correctly then you could use invoked callbacks like this:
```js
on: {
  UPDATE_CONTEXT: assign((ctx, ev) => ({ ...ctx, ...ev.update }))
},
invoke: {
  src: () => (sendBack) => {
    source.addEventListener('event', (ev) => {
      sendBack({ type: 'UPDATE_CONTEXT', update: computeUpdate(ev) })
    })
  }
}
```

**Reply by 000luna000** [reply]
thanks， I do it like this
